src/headers/tomcrypt_pk.h

```diff
@@ -284,18 +298,35 @@ typedef struct {
     void *z;
 } ecc_point;
 
+/** ECC key's domain parameters */
+typedef struct {
+   /** The size of the curve in octets */
+   int size;
+   /** The prime that defines the field the curve is in */
+   void *prime;
+   /** The fields A param */
+   void *A;
+   /** The fields B param */
+   void *B;
+   /** The order of the curve */
+   void *order;
+   /** The base point G on the curve */
+   ecc_point base;
+   /** The co-factor */
+   unsigned long cofactor;
+   /** The OID structure */
+   oid_st oid;
```

Comment:
this will prevent compilation of a user-space app without defining `LTC_SOURCE`

Reply:
The newly introduced `ltc_ecc_dp` was intended to be "invisible" from public API.  User-space apps should neither see nor manipulate this struct directly but only via `ecc_set_dp(..)`.

But I have to check whether this intention really happened.

Reply:
this won't be possible as `ecc_key` contains an instance of `ltc_ecc_dp`

Reply:
You are right. So either moving `oid_st` to public API or switching to `unsigned long oid[16]; unsigned long oidlen;`.